### PR TITLE
Enh/add mkdir p

### PR DESCRIPTION
**Pull request by blakedewey**
Add function to mimic mkdir -p in Linux. Used to correct race condition brought up in parallel execution error from #1127.

### CODE REVIEW

**Comment by blakedewey** [comment]
I'm getting a strange error with `make specs`. Any ideas?

```
Checking specs and autogenerating spec tests
python tools/checkspecs.py
Traceback (most recent call last):
  File "tools/checkspecs.py", line 15, in <module>
    from nipype.external import six
  File "/Users/deweybe/Documents/PythonProjects/nipype/nipype/__init__.py", line 9, in <module>
    from utils.config import NipypeConfig
  File "/Users/deweybe/Documents/PythonProjects/nipype/nipype/utils/config.py", line 20, in <module>
    import filemanip
  File "/Users/deweybe/Documents/PythonProjects/nipype/nipype/utils/filemanip.py", line 20, in <module>
    from ..interfaces.traits_extension import isdefined
  File "/Users/deweybe/Documents/PythonProjects/nipype/nipype/interfaces/__init__.py", line 10, in <module>
    from io import DataGrabber, DataSink, SelectFiles
  File "/Users/deweybe/Documents/PythonProjects/nipype/nipype/interfaces/io.py", line 45, in <module>
    from nipype.interfaces.base import (TraitedSpec, traits, File, Directory,
  File "/Users/deweybe/Documents/PythonProjects/nipype/nipype/interfaces/base.py", line 34, in <module>
    from ..utils.filemanip import (md5, hash_infile, FileNotFoundError,
ImportError: cannot import name hash_infile
make: *** [specs] Error 1
```

File: nipype/utils/filemanip.py [hunk 1]
```diff
@@ -430,3 +431,13 @@ def write_rst_dict(info, prefix=''):
     for key, value in sorted(info.items()):
         out.append(prefix + '* ' + key + ' : ' + str(value))
     return '\n'.join(out)+'\n\n'
+
+
+def mkdir_p(path):
+    try:
+        os.makedirs(path)
```
Comment (satra) [inline nipype/utils/filemanip.py]: i don't think this is an atomic operation, and hence likely to run into the same issue. this is an instance where a lock may be required.

Comment (blakedewey) [inline nipype/utils/filemanip.py]: even with the try is on the creation aspect only? because the only error is EEXISTS according to the initial issue.

**Comment by satra** [comment]
@blakedewey - i think the mkdir bit is fine, but there is a recursive loop because filemanip imports config and config imports filemanip. for this you may want to put the mkdir function in config itself.

**Comment by blakedewey** [comment]
Made the move to config, but I still imported into filemanip, because after all it is a file operation. @satra let me know what you think.

**Pull request merged**